GitUI/CommandsDialogs/FormCommit.cs

```diff
@@ -1089,7 +1089,11 @@ private void ShowChanges(GitItemStatus item, bool staged)
                 return;
             }
 
-            SetSelectedDiff(item, staged);
+            ThreadHelper.JoinableTaskFactory.RunAsync(async () =>
+            {
+                await SetSelectedDiffAsync(item, staged);
```

Comment:
Do we need to call `await this.SwitchToMainThreadAsync();` before?

Reply:
I see no need for this.